File: onlinepayments-sdk/src/main/java/com/onlinepayments/sdk/client/android/model/paymentproduct/validation/Range.java
/*
 * Copyright 2020 Global Collect Services B.V
 */

package com.onlinepayments.sdk.client.android.model.paymentproduct.validation;

import java.io.Serializable;

/**
 * POJO which holds the Range data.
 * Used for validation.
 */
public class Range implements Serializable {

	private static final long serialVersionUID = 4659640500627126711L;

	private Integer minValue;
	private Integer maxValue;

	/**
	 * @deprecated In a future release, this constructor will become internal to the SDK.
	 */
	@Deprecated
	public Range() {}

	public Integer getMinValue(){
		return minValue;
	}

	public Integer getMaxValue(){
		return maxValue;
	}
}

File: onlinepayments-sdk/src/main/kotlin/com/onlinepayments/sdk/client/android/model/paymentproduct/validation/Range.kt
/*
 * Do not remove or alter the notices in this preamble.
 *
 * Copyright © 2025 Worldline and/or its affiliates.
 *
 * All rights reserved. License grant and user rights and obligations according to the applicable license agreement.
 *
 * Please contact Worldline for questions regarding license and user rights.
 */
package com.onlinepayments.sdk.client.android.model.paymentproduct.validation

import java.io.Serializable

/**
 * Data class which holds the Range data.
 * Used for validation.
 */
data class Range(
    val minValue: Int? = null,
    val maxValue: Int? = null
) : Serializable {

    companion object {
        @Suppress("Unused")
        private val serialVersionUID = 4659640500627126711L
    }
}
